### ISSUE TITLE
Hard to read brown text in the crafting menu

### ISSUE THREAD

**Issue by scorpion451** [CleverRaven/Cataclysm-DDA#14143]
Mostly dig the new crafting interface changes, but the brown text is very hard to read. 
To get all "interface optimization and advanced color theory" on ya'll, brown/black has a low perceptive contrast ratio. Which is good for camouflage or subtle color transitions, but very bad for legibility.
I'd suggest  yellow- same color family as the current brown, but much higher contrast.
Alternately, use a brown highlight/background with black text. (Because [math] (https://en.wikipedia.org/wiki/Color_difference) and [eyes] (https://en.wikipedia.org/wiki/Color_constancy))

**Comment by chaosvolt** [comment]
Derp. It would help if you'd dose us with a screenshot of what part, specifically, is the source of this complaint.

**Comment by scorpion451** [comment]
[No problemo ^_^](http://i.imgur.com/p2QnDlG.png)

**Comment by chaosvolt** [comment]
Woah, that's weird. This would be something new, I'm assuming? @_@

**Comment by Nandorianen** [comment]
Meh, you got it easy. Mine are even more subtle.
![cata brown](https://cloud.githubusercontent.com/assets/7211122/11341769/feb45e9c-9214-11e5-8871-2afa76916003.PNG)

**Comment by remyroy** [comment]
A good color profile would be great. Would selectable color profiles be overkill for this? Having multiple interesting profiles and potentially custom ones from the users who prefer something else?

**Comment by BevapDin** [comment]
You can make your own color profile in the color manager (press `?` -> `4`). You can also load color profile (files "data/raw/color_templates"). This feature probably needs more exposure / advertisement.

**Comment by chaosvolt** [comment]
And yet orange will still never exist. :V

**Comment by remyroy** [comment]
Under my current fluorescent lighting, it was hard to read dark blue text on black background (c_blue) which is used for item's material values. Thanks to @BevapDin tip, I changed it to dark blue on a white background (c_blue_white).

**Comment by chaosvolt** [comment]
Hmm. So if I changed it all to plain white...wait. Will this affect anything else, besides the info texts?

Because seriously, the info text option needs to be optional.

**Comment by scorpion451** [comment]
Looked at the color profile option; that's not what I mean. There is nothing wrong with the brown being used- it works great for dirt, trees, etc. The problem is that you are using it for a job it cannot do.

 I am telling you this as someone who lists "technical artist" high on the skills section of my resume: Brown should not be used for menu text at all in this situation. Yellow is a better choice. If you are completely married to the idea that this text must, for some reason, be brown, you could also use black on brown. Brown on black, however, is not conducive to legibility by virtue of physics and how human vision works. That is to say, it ain't gonna work if you want people to be able to read the dang numbers. X3

**Comment by chaosvolt** [comment]
Right. Basically, the default color chosen is a bit odd, I'm hoping that specific instance is being addressed in the PR for it.

**Comment by narc0tiq** [comment]
As I've commented in the past (<https://github.com/CleverRaven/Cataclysm-DDA/pull/14057#issuecomment-157475831>), we're restricted to a 16-color palette whose interpretation is completely up to the terminal displaying it. There is no "brown" in the [ANSI 16-color palette](https://en.wikipedia.org/wiki/ANSI_escape_code#Colors), just "yellow" and "light yellow" -- though some terminals (including Cataclysm's built-in ncurses emulation) tend to use a brownish color for yellow.

For reference, [here's what I see](https://i.imgur.com/WDieUvz.png).

**Comment by Nandorianen** [comment]
If that's the case, tweaking colors used is rather pointless. Changing "yellow" into something else with in-game color editor is also, as noted above, overkill.
This leaves us with externalizing these color options as a proper fix, I suppose? So that we can change color used from "yellow" to "green" or something, and get personally suitable colors.

**Comment by scorpion451** [comment]
We do have yellow among the 16bit colors available. Presumably it is called "light yellow", to judge by your information.
Use that instead of "yellow"-which-is-really-brown-and-has-the-name-proxied-in-game-as-such, particularly if the default appearance is what we are seeing.
Wouldn't mind seeing the externalized controls, either (more customization in an interface is never bad) but I'm guessing that would be a bit tedious to code so it won't work for a quick fix to the problem.

**Comment by narc0tiq** [comment]
Light yellow would probably work better, yes. (Bright) yellow on black is nice and high-contrast.

For the other, I seem to recall there is a way to change color profiles in-game -- `?4` should bring up the color manager, allowing you to remap all instances of a specific color combination to a different one -- however this is still, of course, limited to the normal 16x16 color palette (or possibly even limited to predefined color pairs; I haven't looked at the code to tell which).

**Comment by chaosvolt** [comment]
The thing is, remapping the color palette will affect the map and ASCII tiles, right? So if you'd much rather disable the rainbow vomit, doing so ruins the rest of the game's coloration. -_-

**Comment by scorpion451** [comment]
Wait, record scratch moment here.
This is getting off on a bit of a tangent, but is 16x16 color palette a typo? 4x4 would be 16 colors, 16x16 is 256. Which would blow the door wide open on color...

**Comment by chaosvolt** [comment]
I suspect it's actually the standard 16 colors, plus combinations involving background color.

Which essentially makes it the bootleg 256-color palette. o3o

**Comment by narc0tiq** [comment]
> is 16x16 color palette a typo

16 foreground colors, 16 background colors (though sometimes it's only 8 background colors, because terminals). Same way an 80x24 terminal is not 80 pixels wide and 24 tall (that would be one tiny display!).

**Comment by chaosvolt** [comment]
Exactly. It's TECHNICALLY 256 colors, but only because most of the code references color combos like you'd normally expect colors to be handled.

**Comment by scorpion451** [comment]
Figured it was a computer-logic thing like that. X3
On the graphics side of things when we say 16x16 palette (say, min-maxing the colors in a .gif or prepping an RGB image for CMYK printing) we do mean 256 colors, and dithering is a separate thing. For instance, *really* fancy printers use an analog 5 color palette called CcMmYK (cyan, light cyan, magenta, light magenta, yellow, and key (black/alpha)

What I love is how it gets *really* weird when you dig into the deep layers where everything turns semi-analog: sRGB splits a 4 color analog palette(red, green, blue, and gamma) into a 256x256x256 cube warped over the gamma axis.

**Comment by chaosvolt** [comment]
Though this does imply that my endless yearning to add orange to the game would be a bad idea. ;w;

**Comment by karlnp** [comment]
I still want to get our ncurses emulation away from GDI+ and towards something a little more modern, but it's so goddamn complex it's what turned me off of contributing to cata six months ago.

**Comment by Soadreqm** [comment]
I was going to file an issue on this, but I see there's already one over here. :)

![2015-12-02 21_15_55](https://cloud.githubusercontent.com/assets/6437763/11541973/debbc6e0-993d-11e5-921b-dc9063dca672.png)

This is the default color scheme in the Windows SDL version. On my screen, the brown is annoyingly dark, and the dark blue is almost invisible.
I can edit the colors, but the the brown text is distracting for other reasons. It was used for rotten items, before, and it's acquired some negative connotations for me. I look at the volume, and the first thing in my mind is that it's spoiled. :P

**Comment by kevingranade** [comment]
We are not in control of your terminal colors.
Please set your terminal to use higher contrast colors.

**Comment by kevingranade** [comment]
My bad, if it's sdl it is our fault, probably.

**Comment by chaosvolt** [comment]
The PR that's supposed to unufuck the default colors hasn't been merged yet, so there's the problem. -_-

**Comment by scorpion451** [comment]
One big thing here that doesn't seem to be getting across clearly is that the color brown is not what we're really having trouble with.

Think of it this way; what is happening here is akin to introducing something which caused a portion of the user base to have all of the tiles for zombies replaced by tiles for shrubs. The solution would not be to replace the tiles for shrubs with tiles for zombies. Here, the solution is not "change the color of brown you are using", even if that were an option. : /

**Comment by chaosvolt** [comment]
Logically the solution would be to at least include the option to disable that buggy feature until it's been fixed, so that players that get fucked over by it don't have to wait until it's unfucked. >.>

**Comment by kevingranade** [comment]
Do the new colors in the pr fix the issue?
If so, the option is the only thing holding up the fix.
